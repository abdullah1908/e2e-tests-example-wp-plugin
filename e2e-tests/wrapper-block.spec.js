import {
	clickBlockToolbarButton,
	clickButton,
	createNewPost,
	enablePageDialogAccept,
	getEditedPostContent,
	insertBlock,
} from '@wordpress/e2e-test-utils';
import {
	clickElementByText,
	openSidebarPanelWithTitle,
	selectBlockByName,
	selectOption,
} from './helper';

describe( 'Wrapper block', () => {
	beforeAll( async () => {
		enablePageDialogAccept();
	} );
	beforeEach( async () => {
		await createNewPost();
	} );

	it( 'Wrapper block should be available', async () => {
		await insertBlock( 'Wrapper Block' );

		// Check if block was inserted
		expect( await page.$( '[data-type="e2e-tests-example/wrapper-block"]' ) ).not.toBeNull();

		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	it( 'Background color should be applied', async () => {
		await insertBlock( 'Wrapper Block' );
		await selectBlockByName( 'e2e-tests-example/wrapper-block' );

		// Change background color
		await openSidebarPanelWithTitle( 'Background Color' );
		await selectOption( 'Background Color', 'orange' );

		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	it( 'Margin bottom should be applied', async () => {
		await insertBlock( 'Wrapper Block' );
		await selectBlockByName( 'e2e-tests-example/wrapper-block' );

		// Apply margin bottom
		await openSidebarPanelWithTitle( 'Margin bottom' );
		await clickElementByText( 'label', 'Add margin bottom' );

		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	it( 'Alignment should be set', async () => {
		await insertBlock( 'Wrapper Block' );
		await selectBlockByName( 'e2e-tests-example/wrapper-block' );

		// Change alignment
		await clickBlockToolbarButton( 'Change wrapper block alignment' );
		await clickButton( 'Align Text Center' );

		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );
} );
