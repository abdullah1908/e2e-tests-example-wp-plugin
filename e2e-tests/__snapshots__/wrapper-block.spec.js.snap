// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wrapper block Background color should be applied 1`] = `"<!-- wp:e2e-tests-example/wrapper-block {\\"bgColor\\":\\"orange\\"} /-->"`;

exports[`wrapper block Margin bottom should be applied 1`] = `"<!-- wp:e2e-tests-example/wrapper-block {\\"marginBottom\\":true} /-->"`;

exports[`wrapper block Wrapper block should be available 1`] = `"<!-- wp:e2e-tests-example/wrapper-block /-->"`;
